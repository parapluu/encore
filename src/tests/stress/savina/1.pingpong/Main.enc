active class PingActor
  var pingsLeft : int

  def init(count : int, pong : PongActor) : unit
    this.pingsLeft = count
    pong!sendPing(this)
    this.pingsLeft = this.pingsLeft - 1
  end

  def sendPong(pong : PongActor) : unit
    if this.pingsLeft > 0 then
      pong!sendPing(this)
      this.pingsLeft = this.pingsLeft - 1
    end
  end
end

active class PongActor
  var pongCount : int
  def init() : unit
    this.pongCount = 0
  end
  def sendPing(ping : PingActor) : unit
    print("Got ping {}\n", this.pongCount)
    ping!sendPong(this)
    this.pongCount = this.pongCount + 1
  end
end

active class Main
  def main(args : [String]) : unit
    if |args| != 2 then
      print("input required: number of pings not specified\n")
    else
      let
        N = match (args(1)).to_int() with
              case Just(result) =>
                result
              end
              case Nothing =>
                0
              end
            end
      in
        new PingActor(N, new PongActor())
        print("done")
      end
    end
  end
end
