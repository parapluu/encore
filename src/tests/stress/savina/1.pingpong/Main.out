input required: number of pings not specified
